providers/azuredns/azureDnsProvider.go

```diff
@@ -111,39 +121,66 @@ func (a *azureDnsProvider) GetNameservers(domain string) ([]*models.Nameserver,
 	return ns, nil
 }
 
-// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
-func (client *azureDnsProvider) GetZoneRecords(domain string) (models.Records, error) {
-	return nil, fmt.Errorf("not implemented")
-	// This enables the get-zones subcommand.
-	// Implement this by extracting the code from GetDomainCorrections into
-	// a single function.  For most providers this should be relatively easy.
-}
+func (a *azureDnsProvider) ListZones() ([]string, error) {
+	zonesResult, err := a.getExistingZones()
 
-func (a *azureDnsProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
-	err := dc.Punycode()
+	if err != nil {
+		return nil, err
+	}
+
+	var zones []string
+
+	for _, z := range *zonesResult.Value {
+		domain := strings.TrimSuffix(*z.Name, ".")
+		zones = append(zones, domain)
+	}
+
+	return zones, nil
+}
 
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (a *azureDnsProvider) GetZoneRecords(domain string) (models.Records, error) {
+	existingRecords, _, _, err := a.GetExistingRecords(domain)
 	if err != nil {
 		return nil, err
 	}
+	return existingRecords, nil
+}
 
-	var corrections []*models.Correction
-	zone, ok := a.zones[dc.Name]
+func (a *azureDnsProvider) GetExistingRecords(domain string) (models.Records, []*adns.RecordSet, string, error) {
```

Comment:
Does `GetExistingRecords` need to be exported?  It seems like it could be `getExistingRecords`

Reply:
Yep. can be an internal method. Fixed :)